hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestCallerContext.java

```diff
@@ -42,6 +42,37 @@ public void testBuilderAppend() {
         builder.build().getContext());
   }
 
+  @Test
+  public void testBuilderAppendIfAbsent() {
+    Configuration conf = new Configuration();
+    conf.set(HADOOP_CALLER_CONTEXT_SEPARATOR_KEY, "$");
+    CallerContext.Builder builder = new CallerContext.Builder(null, conf);
+    CallerContext context = builder.append("key1", "value1").build();
+    Assert.assertEquals("key1:value1",
+        builder.build().getContext());
+
+    // Append an existed key.
+    builder.appendIfAbsent("key1", "value1");
```

Comment:
We might want to change the value to make sure that this entry does not overwrite an existing entry which has the same key.
```suggestion
    builder.appendIfAbsent("key1", "value2");
```

Reply:
Thanks @tasanuma for your review and comments. I'll fix it quickly.